integration/pull_test.go

```diff
@@ -1086,3 +1086,54 @@ func addConfig(t *testing.T, sh *shell.Shell, conf string, cmds ...string) []str
 	}
 	return append(cmds, "--config", configPath)
 }
+
+// TestRpullImageThenRemove pulls and rpulls an image then removes it to confirm fuse mounts are unmounted
+func TestRpullImageThenRemove(t *testing.T) {
+	regConfig := newRegistryConfig()
+	sh, done := newShellWithRegistry(t, regConfig)
+	defer done()
+
+	// Initialize config files for containerd and snapshotter
+	additionalConfig := ""
+	if !isTestingBuiltinSnapshotter() {
+		additionalConfig = proxySnapshotterConfig
+	}
+	containerdConfigYaml := testutil.ApplyTextTemplate(t, `
+version = 2
+
+[plugins."io.containerd.snapshotter.v1.soci"]
+root_path = "/var/lib/soci-snapshotter-grpc/"
+
+[plugins."io.containerd.snapshotter.v1.soci".blob]
+check_always = true
+
+{{.AdditionalConfig}}
+`, struct {
+		AdditionalConfig string
+	}{
+		AdditionalConfig: additionalConfig,
+	})
+
+	// Setup environment
+	if err := testutil.WriteFileContents(sh, defaultContainerdConfigPath, []byte(containerdConfigYaml), 0600); err != nil {
+		t.Fatalf("failed to write %v: %v", defaultContainerdConfigPath, err)
+	}
+	sh.
+		X("update-ca-certificates").
+		Retry(100, "nerdctl", "login", "-u", regConfig.user, "-p", regConfig.pass, regConfig.host)
+
+	rebootContainerd(t, sh, "", "")
+
+	containerImage := nginxImage
+
+	copyImage(sh, dockerhub(containerImage), regConfig.mirror(containerImage))
+	indexDigest := optimizeImage(sh, regConfig.mirror(containerImage))
+	sh.X("soci", "image", "rpull", "--user", regConfig.creds(), "--soci-index-digest", indexDigest, regConfig.mirror(containerImage).ref)
+
+	checkFuseMounts(t, sh, 6)
```

Comment:
Magic number could be calculated from the manifest for the downloaded image. Is that desired here?

Reply:
In the case w/o sparse index, yes. However, the current state of snapshotter has the `min-layer-size` set to non-zero value, which implies having a sparse index. This magic number can still be calculated from manifest, but you will need to specify `min-layer-size` explicitly when building soci index for that test.

Reply:
Could you please elaborate on this?

> the current state of snapshotter has the min-layer-size set to non-zero value

Reply:
We skip creating ztocs for layers that are smaller than `min-layer-size` (currently 10 MiB, I believe) so we might be wrong if we use the image manifest to figure out how many fuse mounts to expect.

Since the index iteself is the source of truth, I think we should get the number of ztocs from the index and verify that there are that many FUSE mounts.

Sort of like this:
https://github.com/awslabs/soci-snapshotter/blob/main/integration/index_test.go#L83-L87

We should probably also verify that there is at least 1 ztoc to make sure we don't accidentally raise the default min-layer-size so high that this test skips all layers.

Reply:
10<<20 bytes is the default, but optimizeImage specifies a min layer size of 0 when calling buildSparseIndex (which should produce a non-sparse index?)